x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -107,41 +110,49 @@ mod tests {
         let owner_key = String::from("owner");
         let bob_key = Param::Key(Key::Ed25519(String::from("bob")));
 
-        let mut plan = Plan::new(&owner_key);
-
-        plan.add_step(Step::create_key(Key::Ed25519(owner_key.clone())));
-
-        plan.add_step(Step {
-            endpoint: Endpoint::Key,
-            method: "key_create".into(),
-            params: vec![bob_key.clone()],
-            max_units: 0,
-        });
-
-        let counter_id = plan.add_step(Step {
-            endpoint: Endpoint::Execute,
-            method: "program_create".into(),
-            max_units: 1000000,
-            params: vec![Param::String(PROGRAM_PATH.into())],
-        });
-
-        plan.add_step(Step {
-            endpoint: Endpoint::Execute,
-            method: "inc".into(),
-            max_units: 1000000,
-            params: vec![counter_id.into(), bob_key.clone(), 10.into()],
-        });
-
-        let plan_responses = simulator.run_plan(plan).unwrap();
-
-        assert!(
-            plan_responses.iter().all(|resp| resp.base.error.is_none()),
-            "error: {:?}",
-            plan_responses
-                .iter()
-                .filter_map(|resp| resp.base.error.as_ref())
-                .next()
-        );
+        simulator
+            .run_step::<()>(
+                &owner_key,
+                &Step::create_key(Key::Ed25519(owner_key.clone())),
+            )
+            .unwrap();
```

Comment:
These unwraps are also going aren't going to give the best message for failing tests either. You should make this test return a `Result` as well.

Reply:
I'm going to leave the `unwrap()` since they actually provide a better error message than short-circuiting by the use of `?` and returning a `Result` from the test, because it gives us the backtrace with the line where it panicked in case it happens. It is going to give us an easier time to debug if any of the tests fail since we will exactly know where is failed.